### ISSUE TITLE
New UI controls incomplete / not working

### ISSUE THREAD

**Issue by phoenixtechnam** [hristo-atanasov/Tasmota-IRHVAC#131]
Since HA update to 2023.12.0 my airconditioners based on IRHVAC cannot be controlled properly anymore. Target temperature is not displaying and cannot be be set via UI. See screenshot below:

![image](https://github.com/hristo-atanasov/Tasmota-IRHVAC/assets/29205326/8e0c427d-c398-45a4-8441-73581f961286)

**Comment by cociweb** [comment]
Similar to this:
https://github.com/home-assistant/frontend/issues/18950

**Comment by phoenixtechnam** [comment]
yes, but the problem is not the missing modes, those can be added via additional card configuration. The real problem is that I cannot set target temperatures anymore. It also seems to mix up the current temperature and the target temperature.

**Comment by cociweb** [comment]
As You can see in the legacy ticket, the target temp. is also missing.

**Comment by cociweb** [comment]
well, I was aware to reproduce by a faulty formed json: if you are not initialize the value, and the default value is null, then it can happen that you miss the +/- buttons and you are not able set the temp. <= Since you are not able to set any target temp, then you will get back null. :D  - vicious circle...
As a workaround try to set it at least once with your IR remote, or send a correctly formed mqtt message for tasmota.

**Comment by cociweb** [comment]
According to this, there is another way of thinking in my mind...
What if....
if you are not able to set the temp by the thermostat card....
from your default configuration temperature should derive. Have you set default temperature in your [yaml config](https://github.com/hristo-atanasov/Tasmota-IRHVAC/blob/dca493da6f3680a5a753840b2bae440b6c482343/examples/configuration.yaml#L22)? anyways, it should be set other than 'null' by default....
So 2 option can pops up:
a) the new thermostat card overwrites the default target_temp value with 'null'
b) the default values are not working somehow as [here](https://github.com/hristo-atanasov/Tasmota-IRHVAC/issues/129) reported

**Comment by phoenixtechnam** [comment]
> According to this, there is another way of thinking in my mind... What if.... if you are not able to set the temp by the thermostat card.... from your default configuration temperature should derive. Have you set default temperature in your [yaml config](https://github.com/hristo-atanasov/Tasmota-IRHVAC/blob/dca493da6f3680a5a753840b2bae440b6c482343/examples/configuration.yaml#L22)? anyways, it should be set other than 'null' by default.... So 3 option can pops up: a) the new thermostat card overwrites the default target_temp value with 'null' b) the default values are not working somehow as [here](https://github.com/hristo-atanasov/Tasmota-IRHVAC/issues/129) reported c) somehow Tasmota returns 'null' value instead of None and it is not handled well [here](https://github.com/hristo-atanasov/Tasmota-IRHVAC/blob/dca493da6f3680a5a753840b2bae440b6c482343/custom_components/tasmota_irhvac/climate.py#L644C1-L649C64)
> 
> > Eg:
> > ```
> > {"StateMode": "SendStore", 
> > ....
> > "Power": "off", 
> > "Mode": "off", "
> > Celsius": "on", 
> > "Temp": null, 
> > "FanSpeed": "max",
> > "SwingV": "auto", 
> > "SwingH": "off", 
> > ...}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Instead of:
> > ```
> > {"StateMode": "SendStore", 
> > ....
> > "Power": "off", 
> > "Mode": "off", "
> > Celsius": "on", 
> > "Temp": 25, 
> > "FanSpeed": "max",
> > "SwingV": "auto", 
> > "SwingH": "off", 
> > ...}
> > ```
> 
> Some developer opinion would be appreciated!

min_temp, max_temp and target_temp are all set in my config

**Comment by reloadxero** [comment]
here is what i did to restore functionality, you only have to do it once per entity:

1. In HA go to "Developer Tools"
2. Select "Services" tab
3. Select service "Climate: Set target temperature"
4. Select target climate entity and enter a temperature (any temperature is ok)
5. Hit "Call Service"

As soon as you make this service call the thermostat entity is going to get the target temperature information, the plus and minus buttons as well as full functionality.

**Comment by BigHomie90** [comment]
Thank you @reloadxero ! Even after HA restart, works as it should.

**Comment by phoenixtechnam** [comment]
Thank you @reloadxero, confirmed working

**Comment by Michel01975** [comment]
Thank you @reloadxero

**Comment by parcox** [comment]
> here is what i did to restore functionality, you only have to do it once per entity:
> 
> 1. In HA go to "Developer Tools"
> 2. Select "Services" tab
> 3. Select service "Climate: Set target temperature"
> 4. Select target climate entity and enter a temperature (any temperature is ok)
> 5. Hit "Call Service"
> 
> As soon as you make this service call the thermostat entity is going to get the target temperature information, the plus and minus buttons as well as full functionality.

Thank you @reloadxero, it's working. FYI, I have 3 tasmota-irhvac configured devices, one of them had this issue after HA 2024.2 update.

**Comment by cociweb** [comment]
> here is what i did to restore functionality, you only have to do it once per entity:
> 
> 1. In HA go to "Developer Tools"
> 2. Select "Services" tab
> 3. Select service "Climate: Set target temperature"
> 4. Select target climate entity and enter a temperature (any temperature is ok)
> 5. Hit "Call Service"
> 
> As soon as you make this service call the thermostat entity is going to get the target temperature information, the plus and minus buttons as well as full functionality.

@nao-pon Do you have any suggestion how can we resolve it in the code during __init__? (I would like to avoid to send any/fake temperature to the tasmota after HA restart, because my devices are beeping after any command, and it is annoying)

**Comment by nao-pon** [comment]
For some reason, it seems that the saved target temperature cannot be restored. I'm not sure if this will affect it, but I'm making some modifications to the master, so can I try it out with the current master first to see how it behaves?

You can install it by selecting master in the HACS redownload selection.

**Comment by cociweb** [comment]
> I'm not sure if this will affect it, but I'm making some modifications to the master

Oh my holly cow! It is initialized properly! So, now the Temperature attribute is not null anymore after HA restart/update which resulted in a temperature ping-ping where the ball was a huge null. :D
Thank you! - eagerly waiting for the released version, to close this long-runner
